### PR TITLE
Pin Device Key regeneration To <2.5

### PR DESCRIPTION
**Pull request by strseb**
So 2 Things go wrong here: 

We're refreshing the device key on any update, even though we only need to to it when a key has been used before the 2.5 update. 

Also looking at the video at: 0:51-0:57.  https://mozilla-hub.atlassian.net/browse/VPN-1747
- After refresh  we also scheudle a full account refresh, pulling in the new devices
- We can see on the video the device count now dropped from 5 to 4
- I think the update of the device failed, and thus guardian has removed it. (idk why tho)
- It's likely that the devicemodel no longer includes the current keys. 
- So we should check the nullptr case on activation this is the case. Otherwise we will crash once we deref the device :)

Maybe we should force a full reset() of the app here, so the user can-re login and reauth the device.

### CODE REVIEW

File: src/controller.cpp [hunk 1]
```diff
@@ -291,6 +291,12 @@ void Controller::activateInternal() {
 void Controller::activateNext() {
   MozillaVPN* vpn = MozillaVPN::instance();
   const Device* device = vpn->deviceModel()->currentDevice(vpn->keys());
+  if (device == nullptr) {
```
Comment (bakulf) [inline src/controller.cpp:294]: how can it be we are here? Can you tell me more?

**Review by bakulf — CHANGES_REQUESTED** [review]

**Comment by bakulf** [comment]
>     * After refresh  we also scheudle a full account refresh, pulling in the new devices

Right.

>     * I think the update of the device failed, and thus guardian has removed it. (idk why tho)

This is now shown in the log. In the log we see a full activation after the device key re-generation.

> Maybe we should force a full reset() of the app here, so the user can-re login and reauth the device.

Good point. Follow up?
Let's change this PR to stop the key regeneration at each update.

**Comment by strseb** [comment]
Followup for the controllerstuff is here https://github.com/mozilla-mobile/mozilla-vpn-client/issues/2776

**Review by bakulf — APPROVED** [review]

**Comment by bakulf** [comment]
I merge this and I cherry-pick it to 2.7.1 too

**Pull request merged**